### PR TITLE
test: add snapshots to graph related tests also disable chromatic on geneset names and tooltips

### PR DESCRIPTION
**Pull request by seve**
#716 #760 

Adds a util to take a snapshot of the graph for doing visual diff testing. Also prevents chromatic from recognizing the geneset names as diffs

### CODE REVIEW

File: client/__tests__/e2e/e2e.test.ts [hunk 1]
```diff
@@ -609,13 +616,14 @@ for (const option of options) {
...
     });
-    test("color by mean expression", async ({ page }) => {
+    test("color by mean expression", async ({ page }, testInfo) => {
       await setup(option, page);
       await createGeneset(meanExpressionBrushGenesetName, page);
       await addGeneToSetAndExpand(meanExpressionBrushGenesetName, "SIK1", page);
 
       await colorByGeneset(meanExpressionBrushGenesetName, page);
       await assertColorLegendLabel(meanExpressionBrushGenesetName, page);
+      snapshotTestGraph(page, testInfo);
```
Comment (seve) [inline client/__tests__/e2e/e2e.test.ts:626]: adding snapshot

File: client/__tests__/e2e/e2e.test.ts [hunk 2]
```diff
@@ -379,6 +385,7 @@ describe("ui elements don't error", () => {
     await drag("layout-graph", panCoords.start, panCoords.end, page);
 
     await page.evaluate("window.scrollBy(0, 1000);");
+    snapshotTestGraph(page, testInfo);
```
Comment (seve) [inline client/__tests__/e2e/e2e.test.ts:388]: adding snapshot

File: client/__tests__/e2e/e2e.test.ts
Comment (seve) [inline client/__tests__/e2e/e2e.test.ts:1]: looks prettier ran its course on the file, ive commented the actual diffs

File: client/src/components/util/truncate.tsx [hunk 3]
```diff
@@ -92,7 +92,8 @@ export default (props: any) => {
   // we need an ID to check for this content, since this is the only place the geneset description appears
   const descriptionContent = (
     <span
-      test-id={`geneset-description-tooltip-${originalString}${tooltipAddendum}`}
+      data-testid={`geneset-description-tooltip-${originalString}${tooltipAddendum}`}
+      data-chromatic="ignore"
```
Comment (seve) [inline client/src/components/util/truncate.tsx:96]: chromatic ignore

File: client/src/components/geneExpression/geneSet.tsx [hunk 4]
```diff
@@ -128,6 +128,7 @@ class GeneSet extends React.Component<{}, State> {
                   maxWidth: globals.leftSidebarWidth - genesetNameLengthVisible,
                 }}
                 data-testid={`${setName}:geneset-name`}
+                data-chromatic="ignore"
```
Comment (seve) [inline client/src/components/geneExpression/geneSet.tsx:131]: chromatic ignore

File: client/__tests__/e2e/cellxgeneActions.ts [hunk 5]
```diff
@@ -643,18 +644,32 @@ export async function assertUndoRedo(
...
+    { page }
+  );
+}
+
+export async function snapshotTestGraph(page: Page, testInfo: TestInfo) {
+  const buttonID = "capture-and-display-graph";
+  const imageID = "graph-image";
+  await page.getByTestId(buttonID).click();
+  page.getByTestId(imageID).waitFor();
+  await takeSnapshot(page, testInfo);
```
[the remark this answers is not in the document]
Comment (tihuan) [reply client/__tests__/e2e/cellxgeneActions.ts:672]: that's correct! Chromatic saves the latest approved snapshot in its own DB to diff with each PR!

**Review by tihuan — COMMENTED** [review]
Code looks good! Thanks for setting this up, @seve 🎉 

Probably just need to pass smoke test, so we can see the new diff in Chromatic 🤞

**Review by tihuan — APPROVED** [review]
Sorry didn't LGTM it last time lol

**Pull request merged**